design/authenticators/authenticators_as_external_plugins.md

```diff
@@ -0,0 +1,36 @@
+# Solution Design - Authenticator as a Plugin
+
+## Table of Contents
+
+- [Problem Statement](#problemstatement)
+- [Goals](#goals)
+- [Implementation Steps](#implementationsteps)
+
+
+## Problem Statement
+
+Currently adding new authenticator type in Conjur / DAP requires code changes in Conjur.
+
+Thus is only possible to external users by code contribution to Conjur, and impossible for a concrete DAP version (which uses predefined Conjur version)
+
+Adding an ability to implement authenticator as an external plugin will allow our customers to implement authenticators of their own without changing Conjur codebase 
+
+and without being familiar with Ruby / Conjur
```

Comment:
can the plugins be written in a language other than Ruby?

Reply:
I will give the whole thing some more thought soon, but I do want to second this.

I think the value of plugins for this is greatly increased if they can be implemented in another language of the end-user's choosing. (Note, it's possible that from Conjur's perspective it could still be "all Ruby" through the use of foreign function interfaces or similar).

Reply:
Agree, thats what I planned also